nixos/modules/services/misc/ollama.nix

```diff
@@ -1,71 +1,72 @@
 { config, lib, pkgs, ... }:
 let
-  inherit (lib) types mkBefore;
+  inherit (lib) literalExpression types mkBefore;
 
   cfg = config.services.ollama;
   ollamaPackage = cfg.package.override {
     inherit (cfg) acceleration;
   };
+
+  staticUser = cfg.user != null && cfg.group != null;
 in
 {
   imports = [
     (lib.mkRemovedOptionModule [ "services" "ollama" "listenAddress" ]
       "Use `services.ollama.host` and `services.ollama.port` instead.")
+    (lib.mkRemovedOptionModule [ "services" "ollama" "sandbox" ]
+      "Set `services.ollama.user` and `services.ollama.group` instead.")
+    (lib.mkRemovedOptionModule [ "services" "ollama" "writablePaths" ]
+      "The models directory is now always writable." )
   ];
 
   options = {
     services.ollama = {
       enable = lib.mkEnableOption "ollama server for local large language models";
       package = lib.mkPackageOption pkgs "ollama" { };
+
+      user = lib.mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "ollama";
+        description = ''
+          User account under which to run ollama. Defaults to [`DynamicUser`](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=)
+          when set to `null`.
+
+          The user will automatically be created, if this option is set to a non-null value.
+        '';
+      };
+
+      group = lib.mkOption {
+        type = with types; nullOr str;
+        default = cfg.user;
+        defaultText = literalExpression "services.ollama.user";
+        example = "ollama";
+        description = ''
+          Group under which to run ollama. Only used when `services.ollama.user` is set.
+
+          The group will automatically be created, if this option is set to a non-null value.
+        '';
+      };
+
       home = lib.mkOption {
         type = types.str;
-        default = "%S/ollama";
+        default = "/var/lib//ollama";
```

Comment:
Looks like you left a double slash here, I presume accidentally.

Also, why not use `default` and `defaultText` like you did elsewhere?

Reply:
Because I reuse `home` for users.users.<name>.home and the format string is specific to the systemd unit.

Reply:
I don't understand either.

Reply:
Don't understand what?

Reply:
Your explanation on this.

Reply:
Is it clearer with the excerpt?

Reply:
OK, thank you.

Reply:
Then shouldn't `models` be changed to be based on `home` too?

``` diff
 models = lib.mkOption {
   type = types.str;
-  default = "%S/ollama/models";
-  defaultText = "/var/lib/ollama/models";
+  default = "${cfg.home}/models";
+  defaultText = "$HOME/models";
```

Reply:
Sure, why not.